apis/apps/v1beta4/rebalance_types.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta4
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RebalanceSpec represents the desired spec of Rebalance
+type RebalanceSpec struct {
+	// +kubebuilder:validation:required
+	// InstanceName represents the name of EmqxEnterprise CR
+	InstanceName string `json:"instanceName,omitempty"`
+	// +kubebuilder:validation:required
+	// RebalanceStrategy represents the strategy of EMQX rebalancing
+	// More info: https://docs.emqx.com/en/enterprise/v4.4/advanced/rebalancing.html#rebalancing
+	RebalanceStrategy *RebalanceStrategy `json:"rebalanceStrategy,omitempty"`
```

Comment:
I think this type doesn't user pointer, because the `RebalanceStrategy ` will not equal nil